pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -175,8 +175,9 @@ func (e *Controller) Run(ctx context.Context, workers int) {
 
 	defer e.queue.ShutDown()
 
-	klog.Infof("Starting endpoint controller")
-	defer klog.Infof("Shutting down endpoint controller")
+	logger := klog.LoggerWithName(klog.FromContext(ctx), "Endpoint")
```

Comment:
Elsewhere we decided to use WithName in cmd/kube-controller-manager/app/core.go  where it then applies to New and Run. Please move this and use "endpoint" (lowercase) as name for consistency reasons.

/lgtm cancel